sky/serve/serve_utils.py

```diff
@@ -808,10 +815,11 @@ def _format_replica_table(replica_records: List[Dict[str, Any]],
         replica_records = replica_records[:_REPLICA_TRUNC_NUM]
 
     for record in replica_records:
+        endpoint = record.get('endpoint', None)
         service_name = record['service_name']
         replica_id = record['replica_id']
         version = (record['version'] if 'version' in record else '-')
-        replica_ip = '-'
+        replica_ip = endpoint if endpoint else '-'
```

Comment:
How about removing this `replica_ip` and using the `endpoint` (rename to `replica_endpoint`) above below?

Reply:
`record['endpoint']` can be `None` if the handle did not exist or the port was not yet exposed, so I had split the logic into [getting the endpoint](https://github.com/skypilot-org/skypilot/pull/3377#discussion_r1545791568), and then changing it to `-` if it was `None` (could be `None` from the payload or the default value set above).

Another way to do this could be:
```
endpoint = record.get('endpoint', None) if record.get('endpoint', None) else '-' 
```
but that seemed a little verbose

Reply:
I have renamed `replica_ip` to `replica_endpoint` if that helps